ethportal-api/src/types/content_key/state.rs

```diff
@@ -268,24 +252,95 @@ mod test {
         );
     }
 
-    /// Asserts that encoding and decoding returns the same key.
-    fn assert_encode_decode(key: &StateContentKey) {
-        let bytes = key.to_bytes();
-        let decoded_key = StateContentKey::try_from(bytes).unwrap();
-        assert_eq!(key, &decoded_key);
+    #[rstest]
+    #[case::account_trie_node_key("account_trie_node_key.yaml")]
+    #[case::contract_storage_trie_node_key("contract_storage_trie_node_key.yaml")]
+    #[case::contract_bytecode_key("contract_bytecode_key.yaml")]
+    fn encode_decode(#[case] filename: &str) -> Result<()> {
```

Comment:
I don't think that we need to return `Result` here and below., because we are not doing anything with the error. In tests we usually want to panic or return `( )`.

Reply:
I personally like to return `Result` in tests because I can use `?` operator instead of `.unwrap()`, which in my mind makes code cleaner (returning Err makes the code fail).

I also don't think it's something that is generally considered bad practice. See:
Rust Book: https://doc.rust-lang.org/book/ch11-01-writing-tests.html#using-resultt-e-in-tests
Rust by Example: https://doc.rust-lang.org/rust-by-example/testing/unit_testing.html#tests-and-